drivers/can/can_stm32.c

```diff
@@ -409,6 +410,46 @@ static int can_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+	/* configure pinmux */
+	if (cfg->pinctrl_len != 0U) {
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
+		int remap;
+		/* Check that remap configuration is coherent across pins */
+		remap = stm32_dt_pinctrl_remap_check(cfg->pinctrl,
+						     cfg->pinctrl_len);
+		if (remap < 0) {
+			return remap;
+		}
+
+		/* A valid remapping configuration is provided */
+		/* Apply remapping before proceeding with pin configuration */
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+
+		switch ((uint32_t)cfg->can) {
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(can1), okay)
+		case DT_REG_ADDR(DT_NODELABEL(can1)):
+			if (remap == REMAP_FULL) { /* or REMAP_1 ??? */
+				LL_GPIO_AF_EnableRemap_CAN1();
+			} else {
+				LL_GPIO_AF_DisableRemap_CAN1();
+			}
```

Comment:
So on CAN1, you actually the following:
NO_REMAP (PA11/PA12)
LL_GPIO_AF_RemapPartial1_CAN1
REMAP_1 (PB8/PB9)
LL_GPIO_AF_RemapPartial2_CAN1
REMAP_2 (PD0/PD1)
LL_GPIO_AF_RemapPartial3_CAN1